app/src/main/java/com/jHerscu/clearskies/di/NetworkModule.kt

```diff
@@ -1,11 +1,86 @@
 package com.jHerscu.clearskies.di
 
+import com.jHerscu.clearskies.BuildConfig
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
 import dagger.Module
+import dagger.Provides
 import dagger.hilt.InstallIn
 import dagger.hilt.components.SingletonComponent
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import javax.inject.Named
+import javax.inject.Singleton
 
+private const val BASE_WEATHER_URL = "https://api.openweathermap.org/"
+
+private const val CURRENT_WEATHER_URL = "data/2.5/onecall"
+
+private const val HISTORIC_WEATHER_URL = "/timemachine"
+
+private const val IMAGE_WEATHER_URL = "img/wn/"
+
+private const val GEOCODE_URL = "https://api.radar.io/v1/search/autocomplete"
 
 @Module
 @InstallIn(SingletonComponent::class)
 object NetworkModule {
+
+    @Provides
+    @Named("weather_key")
+    fun provideWeatherApiKey(): String {
+        return BuildConfig.OPEN_WEATHER_KEY
+    }
+
+    @Provides
+    @Named("geocode_key")
+    fun provideGeocodeApiKey(): String {
```

Comment:
this isn't really wrong, but it's kind of a leaky detail. Nothing outside of this module class really needs to know about geocode_key or weather_key - so it might make more sense to not expose that to the global dependency graph when they're only used in this class

Reply:
Oh wow, yeah, I can see how that might be a problem.